artiq/frontend/artiq_coreanalyzer.py

```diff
@@ -64,8 +67,12 @@ def main():
                                 decoded_dump,
                                 uniform_interval=args.vcd_uniform_interval)
     if args.write_dump:
+        # normalize the dump endian
+        parts = struct.unpack(endian + "IQbbb", dump[:15])
+        header = struct.pack(">IQbbb", *parts)
```

Comment:
difficult to maintain and breaks abstraction layers, not necessary if endianness processing is moved to ``decode_dump`` as suggested